cloud/services/routetables/routetables.go

```diff
@@ -86,7 +77,7 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 		return errors.New("Invalid Route Table Specification")
 	}
 	klog.V(2).Infof("deleting route table %s", routeTableSpec.Name)
-	future, err := s.Client.Delete(ctx, s.Scope.AzureCluster.Spec.ResourceGroup, routeTableSpec.Name)
+	err := s.Client.Delete(ctx, s.Scope.AzureCluster.Spec.ResourceGroup, routeTableSpec.Name)
```

Comment:
This looks so much cleaner than exposing the future upward from the go-sdk impl.